src/blog/understanding-ecmascript-part-3.md

```diff
@@ -0,0 +1,320 @@
+---
+title: 'Understanding the ECMAScript spec, part 3'
+author: '[Marja Hölttä](https://twitter.com/marjakh), speculative specification spectator'
+avatars:
+  - marja-holtta
+date: 2020-02-11 13:33:37
+tags:
+  - ECMAScript
+description: 'Tutorial on reading the ECMAScript specification'
+tweet: ''
+---
+
+... where we dive deep in the syntax!
+
+## Previous episodes
+
+In [part 2](/blog/understanding-ecmascript-part-2), we examined a simple grammar production and how its runtime semantics are defined.  In [the extra content](/blog/extra/understanding-ecmascript-part-2-extra), we also followed a long grammar production chain from `AssignmentExpression` to `MemberExpression`. In this episode, we'll go deeper in the definition of the ECMAScript (or JavaScript) language and its syntax.
+
+If you're not familiar with [context-free grammars](https://en.wikipedia.org/wiki/Context-free_grammar), now it's a good idea to check out the basics, since the spec uses context-free grammars to define the language.
+
+## ECMAScript grammars
+
+ECMAScript source text is a sequence of Unicode code points. Each Unicode code point is an integral value between `U+0000` and `U+10FFFF`. The actual encoding (for example, UTF-8 or UTF-16) is not important &mdash; we assume that the source code has already been converted into a sequence of Unicode code points according to the encoding it was in.
```

Comment:
I don't understand this comment :) I don't want to talk about "letters and symbols" since, what is a letter and a symbol? (A unicode code point!) Also I didn't say unicode is not important, I said the encoding is not important.

Reply:
I guess my comment is mostly bike-shedding. You use terminology I'm not use to, and a level of detail that I don't consider particularly interesting.
Terminology: You write "Unicode code points" instead of "Unicode characters" or just "characters". After years of training I have an understanding what a character is. Why is a different notion used here?
Why is it interesting that Unicode code points are integrals value between `U+0000` and `U+10FFFF`? Do you want to say that there are a lot of them? 
Why is the encoding (UTF-8 or UTF-16) interesting? As @syg writes, maybe you can just link to a wikipedia article and be done with it."

Reply:
As discussed, rewrote this:

The spec defines ECMAScript source text as a sequence of Unicode characters. This means that variable names are not limited to ASCII characters but can also include other Unicode characters, such as emojis. The spec doesn't talk about the actual encoding (for example, UTF-8 or UTF-16). We assume that the source code has already been converted into a sequence of Unicode characters according to the encoding it was in.